rs/p2p/quic_transport/src/connection_handle.rs

```diff
@@ -21,6 +21,26 @@ use crate::{
     ConnId, MessagePriority, MAX_MESSAGE_SIZE_BYTES,
 };
 
+/// Drop guard to send a [`SendStream::reset`] frame on drop. QUINN sends a [`SendStream::finish`] frame by default when dropping a [`SendStream`],
+/// which can lead to the peer receiving the stream thinking a complete message was sent. This guard is used to send a reset frame instead, to signal
+/// that the transmission of the message was cancelled.
+struct SendStreamDropGuard {
+    send_stream: SendStream,
+}
+
+impl SendStreamDropGuard {
+    fn new(send_stream: SendStream) -> Self {
+        Self { send_stream }
+    }
+}
+
+impl Drop for SendStreamDropGuard {
+    fn drop(&mut self) {
+        // fails silently if the stream is already closed.
+        let _ = self.send_stream.reset(VarInt::from_u32(0));
```

Comment:
Note that error code zero means no error. See https://datatracker.ietf.org/doc/html/draft-ietf-quic-transport-03#section-12.3

> QUIC_STREAM_CANCELLED (0x80000006):  The stream was cancelled

Is probably more appropriate.